### PR TITLE
PAY-5085: Updates GraphQL for Payment Services with GooglePay details

### PR DESCRIPTION
**Pull request by danidelcar**
## Purpose of this pull request

This pull request (PR) updates GraphQL for Payment Services with GooglePay details

## Affected pages

- [getPaymentConfig](https://developer.adobe.com/commerce/webapi/graphql/schema/checkout/queries/get-payment-config/)
- [createPaymentOrder](https://developer.adobe.com/commerce/webapi/graphql/schema/checkout/mutations/create-payment-order/)
- [getPaymentSDK](https://developer.adobe.com/commerce/webapi/graphql/schema/checkout/queries/get-payment-sdk/)
- [Introduction](https://developer.adobe.com/commerce/webapi/graphql/payment-services/)
- Updates links in [Klarna](https://developer.adobe.com/commerce/webapi/graphql/payment-methods/klarna/)

## Additional info

JIRA [PAY-5085](https://jira.corp.adobe.com/browse/PAY-5085)

Related to this, there will be a PR that addresses updates to the `commerce-merchant-services` repo, with more updates for the Payment Services docs

### CODE REVIEW

**Review by rubencougil — APPROVED** [review]
🚀

File: src/pages/graphql/schema/checkout/mutations/create-payment-order.md [hunk 1]
```diff
@@ -86,3 +91,7 @@ Attribute |  Data Type | Description
 `id` | String | The unique order ID generated by PayPal
 `mp_order_id` | String | The unique order ID generated in Commerce if Payment Services is enabled after PayPal returns the `id`
 `status` | String | The status of the payment order. See [order status](https://experienceleague.adobe.com/docs/commerce-admin/stores-sales/order-management/orders/order-status.html) for more information
+`amount` | Float | The amount of the payment order
```
Comment (shrielenee) [inline src/pages/graphql/schema/checkout/mutations/create-payment-order.md:94]: I don't love using "the" as previously mentioned, but it's a pattern in this page so I'd leave it.

File: src/pages/graphql/schema/checkout/queries/get-payment-config.md [hunk 2]
```diff
@@ -205,6 +224,26 @@ The following example runs the `getPaymentConfig` query for a `location: CHECKOU
...
+                "google_pay": {
+                    "payment_source": "googlepay",
+                    "code": "payment_services_paypal_google_pay",
+                    "title": "Google Pay",
+                    "payment_intent": "authorize",
+                    "sort_order": "0",
+                    "sdk_params": [
+                        {
+                            "name": "src",
+                            "value": "https://www.paypal.com/sdk/js?client-id=..."
```
Comment (shrielenee) [inline src/pages/graphql/schema/checkout/queries/get-payment-config.md:236]: Is this supposed to be a full URL or was it truncated?

File: src/pages/graphql/schema/checkout/queries/get-payment-config.md [hunk 3]
```diff
@@ -226,6 +265,7 @@ The `PaymentConfigOutput` contains details about each configured payment method:
 Attribute |  Data Type | Description
 --- | --- | ---
 `ApplePayConfig` | String! | ApplePay payment method configuration
+`GooglePayConfig` | String! | GooglePay payment method configuration
```
Comment (shrielenee) [inline src/pages/graphql/schema/checkout/queries/get-payment-config.md:268]: Aren't these supposed to be "Apple Pay payment method configuration" and "Google Pay payment method configuration" and similar? They ar smashed together here.

File: src/pages/graphql/schema/checkout/queries/get-payment-config.md [hunk 4]
```diff
@@ -268,6 +309,31 @@ Attribute | Data Type | Description
...
 
+### `GooglePayConfig` attributes
+
+The `GooglePayConfig` payment method configuration contains the following attributes:
+
+Attribute |  Data Type | Description
+--- | --- | ---
+`code` | String | The payment method code as defined in the payment gateway
+`is_visible` | Boolean | Indicates whether the payment method is shown
+`payment_intent` | String | Defines the payment intent. The possible values are `Authorize` or `Capture`
```
Comment (shrielenee) [inline src/pages/graphql/schema/checkout/queries/get-payment-config.md:320]:
```suggestion
`payment_intent` | String | Defines the payment intent. The possible values are `Authorize` or `Capture`.
```

File: src/pages/graphql/schema/checkout/queries/get-payment-config.md [hunk 5]
```diff
@@ -268,6 +309,31 @@ Attribute | Data Type | Description
...
+`sort_order` | String | The relative order the payment method is displayed on the checkout page
+`title` | String! | The display name of the payment method
+
+The `GooglePayConfig` payment method configuration also has a `button_styles` object containing the following attributes:
+
+Attribute | Data Type | Description
+--- | --- | ---
+`color` | String | The button color
+`height` | Int | The button height in pixels
+`size_mode` | String | Defines the button size mode. The possible values are `static` or `fill`
```
Comment (shrielenee) [inline src/pages/graphql/schema/checkout/queries/get-payment-config.md:332]:
```suggestion
`size_mode` | String | Defines the button size mode. The possible values are `static` or `fill`.
```

File: src/pages/graphql/schema/checkout/queries/get-payment-config.md [hunk 6]
```diff
@@ -268,6 +309,31 @@ Attribute | Data Type | Description
...
+`type` | String | Defines the button type. The possible values are `book`, `buy`, `checkout`, `donate`, `order`, `pay`, `plain`, and `subscribe`
```
Comment (shrielenee) [inline src/pages/graphql/schema/checkout/queries/get-payment-config.md:333]:
```suggestion
`type` | String | Defines the button type. The possible values are `book`, `buy`, `checkout`, `donate`, `order`, `pay`, `plain`, and `subscribe`.
```

**Review by shrielenee — APPROVED** [review]
Some questions but approved with consideration of those

File: src/pages/graphql/schema/checkout/queries/get-payment-config.md (hunk 2 above)
Comment (keharper) [reply src/pages/graphql/schema/checkout/queries/get-payment-config.md:236]: Truncated. You don't want to include sensitive data.

File: src/pages/graphql/schema/checkout/queries/get-payment-config.md (hunk 3 above)
Comment (keharper) [reply src/pages/graphql/schema/checkout/queries/get-payment-config.md:268]: These are names of output objects.

File: src/pages/graphql/schema/checkout/queries/get-payment-config.md (hunk 4 above)
Comment (keharper) [reply src/pages/graphql/schema/checkout/queries/get-payment-config.md:320]: I had to make a lot of compromises along the way. No periods at the end of sentences or sentence fragments. This is consistent throughout the GraphQL reference docs

File: src/pages/graphql/schema/checkout/queries/get-payment-config.md (hunk 5 above)
Comment (keharper) [reply src/pages/graphql/schema/checkout/queries/get-payment-config.md:332]: Stet

File: src/pages/graphql/schema/checkout/queries/get-payment-config.md (hunk 6 above)
Comment (keharper) [reply src/pages/graphql/schema/checkout/queries/get-payment-config.md:333]: Stet

File: src/pages/graphql/schema/checkout/queries/get-payment-config.md (hunk 3 above)
Comment (keharper) [inline src/pages/graphql/schema/checkout/queries/get-payment-config.md:268]:
```suggestion
`GooglePayConfig` | String! | Google Pay payment method configuration
```

**Review by keharper — APPROVED** [review]

File: src/pages/graphql/schema/checkout/queries/get-payment-config.md (hunk 3 above)
Comment (danidelcar) [reply src/pages/graphql/schema/checkout/queries/get-payment-config.md]: done

**Comment by danidelcar** [comment]
> Some questions but approved with consideration of those

thanks!, done couple of them!

File: src/pages/graphql/schema/checkout/queries/get-payment-config.md (hunk 3 above)
Comment (danidelcar) [reply src/pages/graphql/schema/checkout/queries/get-payment-config.md]: done

**Comment by danidelcar** [comment]
Updated PR with latest code changes

**Review by rubencougil — APPROVED** [review]

File: src/pages/graphql/schema/checkout/queries/get-payment-config.md [hunk 7]
```diff
@@ -268,6 +307,30 @@ Attribute | Data Type | Description
...
+`sort_order` | String | The relative order the payment method is displayed on the checkout page
+`title` | String! | The display name of the payment method
+
+The `GooglePayConfig` payment method configuration also has a `button_styles` object containing the following attributes:
+
+Attribute | Data Type | Description
+--- | --- | ---
+`color` | String | The button color
+`height` | Int | The button height in pixels
+`type` | String | Defines the button type. The possible values are `buy`, `checkout`, `order`, `pay`, and `plain`
```
Comment (rubencougil) [inline src/pages/graphql/schema/checkout/queries/get-payment-config.md:330]: 👍

**Review by rubencougil — APPROVED** [review]

File: src/pages/graphql/schema/checkout/queries/get-payment-config.md [hunk 8]
```diff
@@ -276,7 +339,7 @@ Attribute |  Data Type | Description
 --- | --- | ---
 `cc_vault_code` | String | The vault payment method code. Hosted fields only works with credit cards (cc)
 `is_vault_enabled` | Boolean | Indicates whether card vaulting is enabled
-`payment_source` | String | The identifiable payment source for the payment method. The possible values are credit card (cc), PayPal (paypal), and Apple Pay (applepay)
+`payment_source` | String | The identifiable payment source for the payment method. The possible values are credit card (cc), PayPal (paypal), Google Pay (googlepay), and Apple Pay (applepay)
```
Comment (AVoskoboinikov) [inline src/pages/graphql/schema/checkout/queries/get-payment-config.md:342]: For Hoste Fields the only possible value for `payment_source` is "cc".

**Review by AVoskoboinikov — APPROVED** [review]

File: src/pages/graphql/schema/checkout/queries/get-payment-sdk.md [hunk 9]
```diff
@@ -44,33 +44,60 @@ query {
...
-          ]
+    "data": {
+        "getPaymentSDK": {
+            "sdkParams": [
+                {
+                    "code": "payment_services_paypal_smart_buttons",
+                    "params": [
+                        {
+                            "name": "src",
+                            "value": "https:\/\/www.paypal.com\/sdk\/js?client-id=AeFUI2PCqy7MicQ1gLbJf_8T7XUlEG51Pdd7sn2LE3d0_qAVhVBm4VOpPGmV08S9qsMCPXSCiBslhV53&intent=capture&locale=en_US&merchant-id=2PZE7VHLY4B94&currency=USD&commit=false&disable-funding=bancontact,boleto,boletobancario,blik,eps,giropay,ideal,itau,mercadopago,multibanco,mybank,oxxo,payu,p24,satispay,sepa,sofort,trustly,wechatpay,applepay&enable-funding=card,paylater&components=buttons"
```
Comment (AVoskoboinikov) [inline src/pages/graphql/schema/checkout/queries/get-payment-sdk.md:55]: please, delete Ids from here

File: src/pages/graphql/schema/checkout/queries/get-payment-sdk.md [hunk 10]
```diff
@@ -44,33 +44,60 @@ query {
...
+                            "value": "https:\/\/www.paypal.com\/sdk\/js?client-id=AeFUI2PCqy7MicQ1gLbJf_8T7XUlEG51Pdd7sn2LE3d0_qAVhVBm4VOpPGmV08S9qsMCPXSCiBslhV53&intent=capture&locale=en_US&merchant-id=2PZE7VHLY4B94&currency=USD&commit=false&disable-funding=bancontact,boleto,boletobancario,blik,eps,giropay,ideal,itau,mercadopago,multibanco,mybank,oxxo,payu,p24,satispay,sepa,sofort,trustly,wechatpay,credit,card&enable-funding=applepay&components=buttons"
+                        }
+                    ]
+                },
+                {
+                    "code": "payment_services_paypal_google_pay",
+                    "params": [
+                        {
+                            "name": "src",
+                            "value": "https:\/\/www.paypal.com\/sdk\/js?client-id=AeFUI2PCqy7MicQ1gLbJf_8T7XUlEG51Pdd7sn2LE3d0_qAVhVBm4VOpPGmV08S9qsMCPXSCiBslhV53&intent=authorize&locale=en_US&merchant-id=2PZE7VHLY4B94&currency=USD&disable-funding=bancontact,boleto,boletobancario,blik,eps,giropay,ideal,itau,mercadopago,multibanco,mybank,oxxo,payu,p24,satispay,sepa,sofort,trustly,wechatpay&components=googlepay"
```
Comment (AVoskoboinikov) [inline src/pages/graphql/schema/checkout/queries/get-payment-sdk.md:73]: please, delete Ids from here

File: src/pages/graphql/schema/checkout/queries/get-payment-sdk.md [hunk 11]
```diff
@@ -44,33 +44,60 @@ query {
...
+                        },
+                        {
+                            "name": "data-partner-attribution-id",
+                            "value": "MagentoPayments_SP_PCP_Int"
+                        },
+                        {
+                            "name": "data-client-token",
+                            "value": "eyJicmFpbnRyZWUiOnsiYXV0aG9yaXphdGlvbkZpbmdlcnByaW50IjoiMTFiZmFjZGM5YWEyM2ZhZjdmNTQwMzc0NGZmYzEwMjI4YjFjODBmZjg4NDdlYjcyMjMyMmM1OTE0MTU3OWYzZHxtZXJjaGFudF9pZD1yd3dua3FnMnhnNTZobTJuJnB1YmxpY19rZXk9NjNrdm4zN3Z0MjlxYjRkZiZjcmVhdGVkX2F0PTIwMjQtMDItMTNUMTU6Mjc6NTkuNDg5WiIsInZlcnNpb24iOiIzLXBheXBhbCJ9LCJwYXlwYWwiOnsiaWRUb2tlbiI6bnVsbCwiYWNjZXNzVG9rZW4iOiJBMjFBQUlEc1hFbHFlVzh6d1FTTGZsVFdkMHR1UkhIYlBYVTdfYTQzNVFxUVQ5MTRmUjhzclN1RTQzdlg4TnVXV0N1NHZIeUVoVG1BUnhVekdnU3R3VDJCNVFtczBrVjRnIn19"
```
Comment (AVoskoboinikov) [inline src/pages/graphql/schema/checkout/queries/get-payment-sdk.md:90]: is this a valid token?

Comment (danidelcar) [reply src/pages/graphql/schema/checkout/queries/get-payment-sdk.md:90]: I took it from the example wiki where we had all valid examples

File: src/pages/graphql/schema/checkout/queries/get-payment-sdk.md (hunk 9 above)
Comment (danidelcar) [reply src/pages/graphql/schema/checkout/queries/get-payment-sdk.md]: included truncated urls

File: src/pages/graphql/schema/checkout/queries/get-payment-sdk.md [hunk 12]
```diff
@@ -44,33 +44,60 @@ query {
...
+                            "value": "https:\/\/www.paypal.com\/sdk\/js?client-id=AeFUI2PCqy7MicQ1gLbJf_8T7XUlEG51Pdd7sn2LE3d0_qAVhVBm4VOpPGmV08S9qsMCPXSCiBslhV53&intent=capture&locale=en_US&merchant-id=2PZE7VHLY4B94&currency=USD&commit=false&disable-funding=bancontact,boleto,boletobancario,blik,eps,giropay,ideal,itau,mercadopago,multibanco,mybank,oxxo,payu,p24,satispay,sepa,sofort,trustly,wechatpay,applepay&enable-funding=card,paylater&components=buttons"
+                        }
+                    ]
+                },
+                {
+                    "code": "payment_services_paypal_apple_pay",
+                    "params": [
+                        {
+                            "name": "src",
+                            "value": "https:\/\/www.paypal.com\/sdk\/js?client-id=AeFUI2PCqy7MicQ1gLbJf_8T7XUlEG51Pdd7sn2LE3d0_qAVhVBm4VOpPGmV08S9qsMCPXSCiBslhV53&intent=capture&locale=en_US&merchant-id=2PZE7VHLY4B94&currency=USD&commit=false&disable-funding=bancontact,boleto,boletobancario,blik,eps,giropay,ideal,itau,mercadopago,multibanco,mybank,oxxo,payu,p24,satispay,sepa,sofort,trustly,wechatpay,credit,card&enable-funding=applepay&components=buttons"
```
[the remark this answers is not in the document]
Comment (danidelcar) [reply src/pages/graphql/schema/checkout/queries/get-payment-sdk.md]: same

File: src/pages/graphql/schema/checkout/queries/get-payment-sdk.md (hunk 10 above)
Comment (danidelcar) [reply src/pages/graphql/schema/checkout/queries/get-payment-sdk.md]: same

File: src/pages/graphql/schema/checkout/queries/get-payment-sdk.md [hunk 13]
```diff
@@ -44,33 +44,60 @@ query {
...
+                            "value": "https:\/\/www.paypal.com\/sdk\/js?client-id=AeFUI2PCqy7MicQ1gLbJf_8T7XUlEG51Pdd7sn2LE3d0_qAVhVBm4VOpPGmV08S9qsMCPXSCiBslhV53&intent=authorize&locale=en_US&merchant-id=2PZE7VHLY4B94&currency=USD&disable-funding=bancontact,boleto,boletobancario,blik,eps,giropay,ideal,itau,mercadopago,multibanco,mybank,oxxo,payu,p24,satispay,sepa,sofort,trustly,wechatpay&components=googlepay"
+                        }
+                    ]
+                },
+                {
+                    "code": "payment_services_paypal_hosted_fields",
+                    "params": [
+                        {
+                            "name": "src",
+                            "value": "https:\/\/www.paypal.com\/sdk\/js?client-id=AeFUI2PCqy7MicQ1gLbJf_8T7XUlEG51Pdd7sn2LE3d0_qAVhVBm4VOpPGmV08S9qsMCPXSCiBslhV53&intent=capture&locale=en_US&merchant-id=2PZE7VHLY4B94&currency=USD&disable-funding=bancontact,boleto,boletobancario,blik,eps,giropay,ideal,itau,mercadopago,multibanco,mybank,oxxo,payu,p24,satispay,sepa,sofort,trustly,wechatpay&components=hosted-fields"
```
[the remark this answers is not in the document]
Comment (danidelcar) [reply src/pages/graphql/schema/checkout/queries/get-payment-sdk.md]: same

**Pull request merged**